lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,916 @@
+        samplevalue = samplevalue - self:GetTargetPriority(ent)
+
+        if (samplevalue <= 0) then
+            self:SetTarget(ent)
+
+            return true
+        end
+    end
+
+    --slowly die if they end up somewhere where nobody is
+    if (playersum == 0) then
+        self:SetHealth(self:Health() - 5)
+
+        if (self:Health() <= 0) then
+            self:Remove()
+
+            return
+        end
+    end
+
+    self.NextTargetTime = CurTime() + 2
+    self:SetTarget(nil)
+
+    return false
+end
+
+function ENT:ResetBehavior()
+    if (self.Suicidal) then return end
+    self.shifted = nil
+
+    if (self.path and IsValid(self.path)) then
+        self.path:Invalidate()
+        self.path = nil
+    end
+
+    self.ManualTarget = nil
+    self.NeedsTarget = true
+    --self:FindTarget() 
+end
+
+function ENT:WanderToPos(pos)
+    self:ResetBehavior()
+    self.WanderForcePos = type(vector) and pos
+end
+
+function ENT:IgnoreTarget(target)
+    self.TargetBlacklist = self.TargetBlacklist or {}
+    self.TargetBlacklist[target] = CurTime() + 5
+end
+
+function ENT:RunBehaviour()
+    -- This function is called when the entity is first spawned, it acts as a giant loop that will run as long as the NPC exists
+    while (true) do
+        if (self.loco:IsStuck()) then
+            self:HandleStuck()
+        end
+
+        if (self.NeedsTarget) then
+            self:FindTarget()
+            self.NeedsTarget = nil
+        end
+
+        if (IsValid(self:GetTarget()) and self:CanBeTarget(self:GetTarget())) then
+            self.loco:SetDesiredSpeed((self.Suicidal and 350) or (self:GetBased() and 500) or 200)
+            self.loco:SetAcceleration(self.Suicidal and 400 or 200)
+            self.loco:SetDeceleration(self.Suicidal and 400 or 500)
+            self.loco:SetJumpHeight(100)
+
+            if (not self:NearTarget()) then
+                local result = self:ChaseTarget()
+
+                --if chasing the target fails somehow, its probably a wise assumption that its redundant to keep trying.
+                if (result == "failed") then
+                    self:IgnoreTarget(self:GetTarget())
+                    self:ResetBehavior()
+                end
+            end
+
+            if (self:NearTarget()) then
+                if (not self:IsTalking()) then
+                    self:SayStuff()
+                end
+
+                self.loco:FaceTowards(self:GetTarget():GetPos())
+
+                if (not self.shifted) then
+                    local result = self:MoveToPos(self:GetTarget():GetPos() + VectorRand() * Vector(1, 1, 0):GetNormalized() * math.Rand(80, 150), {
+                        maxage = 50
+                    })
+
+                    if (result ~= "ok") then
+                        self:ResetBehavior()
+                    end
+
+                    --cheap method of keeping kleiners with similar targets from clumping together. i figure it's cheaper than some kind of avoidance.
```

Comment:
One thing that might work (it worked for me doing something similar in the past) is, if two kleiners intersect with each other, the one farther away from the target stops until the intersection ends

Reply:
wouldn't that require extra server work doing collisions with eachother